NetworkingLabKenny/app/src/main/java/com/korbkenny/networkinglabkenny/MainActivity.java

```diff
@@ -0,0 +1,187 @@
+
+        Cache cache = new DiskBasedCache(getCacheDir(), 1024*1024);
+        Network network = new BasicNetwork(new HurlStack());
+        mQueue = new RequestQueue(cache,network);
+        mQueue.start();
+
+        mCereal = (Button) findViewById(R.id.cereal_button);
+        mChocolate = (Button) findViewById(R.id.chocolate_button);
+        mTea = (Button) findViewById(R.id.tea_button);
+        mRecyclerView = (RecyclerView) findViewById(R.id.recycler_view);
+
+        LinearLayoutManager linearLayoutManager = new LinearLayoutManager(MainActivity.this, LinearLayoutManager.VERTICAL,false);
+        mRecyclerView.setLayoutManager(linearLayoutManager);
+        mAdapter = new ItemsListAdapter(mList);
+        mRecyclerView.setAdapter(mAdapter);
+
+        mButtons = (LinearLayout) findViewById(R.id.buttons);
+
+
+        ConnectivityManager connectivityManager = (ConnectivityManager) getSystemService(CONNECTIVITY_SERVICE);
+        NetworkInfo networkInfo = connectivityManager.getActiveNetworkInfo();
+
+        if(networkInfo != null && networkInfo.isConnected()){
+            mCereal.setOnClickListener(new View.OnClickListener() {
+                @Override
+                public void onClick(View view) {
+                    performGetVolley(GET_CEREAL_URL);
+                }
+            });
+
+            mChocolate.setOnClickListener(new View.OnClickListener() {
+                @Override
+                public void onClick(View view) {
+                    performGetVolley(GET_CHOCOLATE_URL);
+                }
+            });
+
+            mTea.setOnClickListener(new View.OnClickListener() {
+                @Override
+                public void onClick(View view) {
+                    performGetVolley(GET_TEA_URL);
+                }
+            });
+        }else{
+            mButtons.setOnClickListener(new View.OnClickListener() {
+                @Override
+                public void onClick(View view) {
+                    Toast.makeText(MainActivity.this, "Not connected to the internet...", Toast.LENGTH_SHORT).show();
+                }
+            });
+        }
+    }
+
+    private void performGetVolley(String url){
+        JsonObjectRequest jsonObjectRequest = new JsonObjectRequest(com.android.volley.Request.Method.GET, url,
+                null, new com.android.volley.Response.Listener<JSONObject>() {
+            @Override
+            public void onResponse(JSONObject response) {
+                try {
+                    JSONObject object = new JSONObject(response.toString());
+                    JSONArray listOfProducts = (JSONArray) object.get("items");
+                    mList.clear();
+                    for (int i = 0; i < listOfProducts.length(); i++) {
+                        JSONObject anItem = (JSONObject) listOfProducts.get(i);
+                        mList.add(anItem.getString("name"));
+
+                        MainActivity.this.runOnUiThread(new Runnable() {
```

Comment:
Careful, you don't want to call this inside each iteration of your for loop, just once after the loop